src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,152 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = new Node<T>(value);
+            tail = head;
+        } else {
+            Node<T> newNode = new Node<T>(tail, value);
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        } else {
+            Node<T> current = nodeGetByIndex(index);
+            Node<T> newNode = new Node<T>(current.prev, value, current);
+            if (current == head) {
+                head = newNode;
+            } else {
+                current.prev.next = newNode;
+                current.prev = newNode;
+            }
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexExist(index);
+        Node<T> currentNode = head;
```

Comment:
You can use 'nodeGetByIndex(index)' method to get the currentNode instead of using a loop.